### ISSUE TITLE
Stretched text for labelKey (MouseEvent) | `MOUSE_OUT` -> `gotoAndStop()` causes text to vanish.

### ISSUE THREAD

**Issue by React1-X** [ruffle-rs/ruffle#15110]
### Describe the bug

![buttonissues](https://github.com/ruffle-rs/ruffle/assets/157532981/3dacd27e-2743-437d-b8d4-95833f74e7e7)

![buttonissues2](https://github.com/ruffle-rs/ruffle/assets/157532981/af7ef832-06c6-4f88-84eb-a08cea2092f0)

![image](https://github.com/ruffle-rs/ruffle/assets/157532981/2f291097-8d28-46a5-913a-c88b1438cc57)

Since each of these issues seem closely related, I've included them all in this one issue.
From looking through the code, it seems as though the following are causing each:

1. Text vanishes when `MouseEvent.MOUSE_OUT` is triggered, which uses custom functionality to call `gotoAndStop()`

2. Seems this is either `mouseEvent.target.labelKey` or something to do with system fonts / formats: This code below is what is responsible for the styling of the texts within the buttons:

```javascript
      public function Button()
      {
         super();
         this._startY = y;
         addEventListener(MouseEvent.MOUSE_OVER,this.Over);
         addEventListener(MouseEvent.MOUSE_OUT,this.Out);
         mouseChildren = false;
         buttonMode = true;
         this._format = new TextFormat();
         this._format.font = "Verdana";
         this._format.size = 9;
         this._format.align = TextFormatAlign.CENTER;
         this._format.color = 3355443;
         this._txt = new TextField();
         this._txt.selectable = false;
         this._txt.defaultTextFormat = this._format;
         this._txt.text = "xxxx";
         this._txt.width = 64;
         this._txt.height = 20;
         this.addChild(this._txt);
         this._txt.y = int(height / 2 - this._txt.height / 2 + 2);
         this._txt.x = 1;
         cacheAsBitmap = true;
      }
```

### Expected behavior

For texts to stay visible when `MOUSE_OUT` is triggerd.
For text dimensions to display correctly.
For texts to be visible.

### Content Location

The classes responsible for this behavior:

- [Button.as](https://github.com/bym-refitted/backyard-monsters-refitted/blob/main/client/scripts/Button.as) starting at `line 38`

- [BUILDINGINFO.as](https://github.com/bym-refitted/backyard-monsters-refitted/blob/main/client/scripts/BUILDINGINFO.as) starting at `line 393`

**SWF:** You can find our releases for the SWFs [here](https://github.com/bym-refitted/backyard-monsters-refitted/releases)

You can either connect to our servers and use fake details to bypass the auth, or, grab the localhost file and quickly setup a local server using [these instructions](https://github.com/bym-refitted/backyard-monsters-refitted/wiki/Server-&-Database-Setup).

### Affected platform

Desktop app

### Operating system

Windows 10

**Comment by kjarosh** [comment]
I don't have a way of reproducing this issues, but stretched text should probably be fixed by https://github.com/ruffle-rs/ruffle/pull/22209
